src/main/java/org/gridsuite/modification/server/entities/ModificationEntity.java

```diff
@@ -50,28 +50,53 @@ public class ModificationEntity {
     @Column(name = "modifications_order")
     private int modificationsOrder;
 
+    @Column(name = "message_type")
+    private String messageType;
+
+    @Column(name = "message_values")
+    private String messageValues;
+
     public ModificationEntity(UUID id, ZonedDateTime date, Boolean stashed) {
         this.id = id;
         this.date = date;
         this.stashed = stashed;
     }
 
+    public ModificationEntity(UUID id, ZonedDateTime date, Boolean stashed, String messageType, String messageValues) {
+        this.id = id;
+        this.date = date;
+        this.stashed = stashed;
+        this.messageType = messageType;
+        this.messageValues = messageValues;
+    }
+
     protected ModificationEntity(ModificationInfos modificationInfos) {
         if (modificationInfos == null) {
             throw new NetworkModificationException(MISSING_MODIFICATION_DESCRIPTION, "Missing network modification description");
         }
         //We need to limit the precision to avoid database precision storage limit issue (postgres has a precision of 6 digits while h2 can go to 9)
         this.date = ZonedDateTime.now(ZoneOffset.UTC).truncatedTo(ChronoUnit.MICROS);
+        getModificationMetadata(modificationInfos);
+
     }
 
     public ModificationInfos toModificationInfos() {
         return ModificationInfos.builder()
                 .uuid(this.id)
                 .date(this.date)
                 .stashed(this.stashed)
+                .messageType(this.messageType)
+                .messageValues(this.messageValues)
                 .build();
     }
 
+    public void getModificationMetadata(ModificationInfos modificationInfos) {
+        if (modificationInfos == null) {
+            throw new NullPointerException("Impossible to get label Infos from null DTO");
```

Comment:
You could use `Objects.requireNonNull` and pass the desired message as the second parameter

Reply:
it has has been changed